### DISCUSSION TITLE
AOSP 12.0 v400.h

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2142] [General]
- January Security Patch Level (YES AGAIN)
- Fix boot regression on some devices (notably Samsung ones)

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v400.h'>AOSP 12.0 v400.h</a>.</em>

**Comment by Jeoleo7** [comment]
Voice in the apps other than phone is having high pitched.
ims.apk didn't worked for me unluckily.
I'm using Sony Xperia J9110.

**Reply by Reddy2903** [reply]
Did you try using 'use alternate audio policy' option from treble settings?

**Reply by Jeoleo7** [reply]
Yes, I used the one with the option that says it might fix the audio distortions.

**Reply by Reddy2903** [reply]
Okay. I faced the same issue but it got resolved by selecting that option.

**Comment by tosuman** [comment]
what is actually the floss img package?

**Reply by HorizonsMW** [reply]
floss means "free libre open source software". essentially this variant has open source or free apps pre-installed

**Comment by mrpramodkohli** [comment]
cannot root in samsung a50s 
?? help

**Reply by saujix** [reply]
hey its already rooted
if u try magisk 
the root priviliges will clash
like the error will come 
abnormal root detected blah blah blah

**Comment by Asta-sh** [comment]
Does anyone know how I can install this GSI on my On7xelte?

**Reply by saujix** [reply]
go to twrp 
flash it 
done 
reboot

**Reply by rlucianord** [reply]
if you know how to unlock your bootloader,  then download de apk DeviceInfo   [https://dlandroid.com/device-info-hw-plus-apk](url) ,  Download your GSI  and put it on android tools platform folder(ADB), I did it following this.  [https://source.android.com/setup/build/gsi](url)

**Comment by DavidDeath21** [comment]
problem with the jack I'm using moto G20 please fix that

**Comment by ostapstad** [comment]
Hmm, sorry for stupid question, but how to enable Monet theming? Isn't it included in AOSP?

**Reply by phhusson** [reply]
No, Google still hasn't released it as opensource yet.

**Comment by SteQuOFFICIAL** [comment]
Could someone please help me install this on Google Pixel XL (marlin)?

**Comment by 09kkthxbb** [comment]
Thanks for the update. All are working fine except I can't hear sounds during phone calls, even in loudspeaker mode.

Device: Poco X3 GT (chopin)

**Comment by innofeld** [comment]
Dear Friends,
Between the ROMs, there's "xxx...secure.img.xz"
What does it means exactly? 
Regards

**Reply by devrieshh** [reply]
that you can use your bank apps,its secure

**Reply by innofeld** [reply]
Great
So what kind of features those have, that those are secure for Banking?
Is there any guaranteed system or security patch?
Regards

**Comment by devrieshh** [comment]
they are not rooted and `locked bootloader`( you have to switch to in
development menu)

Op zo 23 jan. 2022 om 18:33 schreef innofeld ***@***.***>:

**Reply by innofeld** [reply]
Dear @devrieshh 
Ah, looks nice
But if it's fully locked, no any trouble or techniques to install on Treble devices?
Can we do change it's locking password by ourselves?
Another matter is Kernel, you know Kernel is so important, some of the Android 12 ROMs have Kernel 5.10.x
So how about Kernels? It'll be upgradable too?
Can we upgrade kernel manually?
Regards
@innofeld

**Comment by devrieshh** [comment]
your phone has to be unlocked,some times you need your phone manufactorer to unlock like xiaomi.
about the kernels see somewere in the discussions till now only the used ones 4.?

**Reply by innofeld** [reply]
Dear @devrieshh
Thanks a lot,
My smartphone is BlackView BV9500plus, Fortunately my device has this feature to "OEM Unlocking".
Regarding to kernel, yes, it has still kernel 4.14.xxx
As you're informed, era of kernels are changing rapidly, and 4.x kernels are outdated; today's kernels are 5.4.xx and 5.10.xx which are more reliable with better overall performance and security with ultimate integrity.
Personally I expect @phhusson if he could integrate his ROMs with kernel 5.4.x and 5.10.x
As I studied in Google sources, Android 12 easily maintainable with kernel 5.10.x

Finally, I have this important question from @phhusson , if we can modify "xxx...secure.img.xz" by our own passwords or not?

Regards

@innofeld

**Comment by devrieshh** [comment]
your phone has to be unlocked,some times you need your phone manufactorer to unlock like xiaomi.
about the kernels see somewere in the discussions till now only the used ones 4.?

**Reply by innofeld** [reply]
Dear @devrieshh
Thanks a lot,
My smartphone is BlackView BV9500plus, Fortunately my device has this feature to "OEM Unlocking".
Regarding to kernel, yes, it has still kernel 4.14.xxx
As you're informed, era of kernels are changing rapidly, and 4.x kernels are outdated; today's kernels are 5.4.xx and 5.10.xx which are more reliable with better overall performance and security with ultimate integrity.
Personally I expect @phhusson if he could integrate his ROMs with kernel 5.4.x and 5.10.x
As I studied in Google sources, Android 12 easily maintainable with kernel 5.10.x

Finally, I have this important question from @phhusson , if we can modify "xxx...secure.img.xz" by our own passwords or not?

Regards

@innofeld

**Comment by tosuman** [comment]
is all the gsi's can't show the location access notification like the camera and microphone access do the notification and the dot also

**Comment by SteQuOFFICIAL** [comment]
Could someone explain to me why there are so many downloads here? Is each file a different version of the system, what do the names of these files mean?

**Reply by solitary-saqieb** [reply]
arm64/arm32_binder64 are architecture types
ab is partition type
floss - only Free and opensource software (no GAPPS)
gapps - google apps
gogapps - google go apps (google go, YouTube go...)
vanilla - no gapps
vndklite - this makes sure system partition is r/w(heard that its easier to update roms?!).

**Reply by innofeld** [reply]
Dear @solitary-saqieb 
I understood, that's cleared.
Some of them are secured, right.
But there's a question, and if those are secured, it means secured/locked by specific password,
So can we modify this password or not? Is that any way to change the password lock by ourselves?
Regards

**Reply by SteQuOFFICIAL** [reply]
@solitary-saqieb Thank You so much!

**Reply by solitary-saqieb** [reply]
@innofeld  

AFAIK secured means it isnt rooted and by passes Google's play protect.
If you download a roooted one then you won't be able to use most of the Google apps unless you hide root and pass safetynet test.

**Comment by devrieshh** [comment]
They are not rooted, accepted as secure by bank

**Reply by innofeld** [reply]
Dear @devrieshh 
I understood, that's cleared.
But there's a question, and if those are secured, it means secured/locked by specific password,
So can we modify this password or not? Is that any way to change the password lock by ourselves?
Regards
@innofeld

**Comment by quoich** [comment]
Hi,

I installed the h version this week and on my phone it says the security patch is December 1st. Was expecting to see January?

Thanks
@quoich

**Reply by phhusson** [reply]
Which variant did you flash?

**Reply by quoich** [reply]
arm64-ab-vndklite-gapps-secure.img.xz

**Reply by phhusson** [reply]
Then that's expected. "secure" variants fakes fingerprint infos to make it look like stock firmware. Which includes security patch.

**Reply by quoich** [reply]
Ok thanks for clarifying. I'm very impressed with it 😊. I replaced MIUI with your AOSP 12.

**Comment by quoich** [comment]
@phhusson When the next update is ready will I be able to do a OTA update via the system updater or will I have to flash the new image?

**Reply by phhusson** [reply]
Asking a volunteer developer for a time is considered inappropriate. (as a rule of thumb, the same is true for paid developers as well)

No OTA at the moment, you'll need to flash it.

**Reply by quoich** [reply]
@phhusson Hi. I wasn't asking for a time. I just wanted to know the update process. Thanks

**Reply by phhusson** [reply]
Ah ok sorry, fair enough.
So yes just flash, but litterally just that, nothing else, so just `fastboot flash system system.img`, nothing to wipe

**Reply by quoich** [reply]
Great, thanks very much

**Comment by Daniel-Richardson-20** [comment]
Hi My Device is POCO M3 Pro 5G. I am not able to use volte in my phone i am able to make calls but can't receive any. I don't why but if someone can help me i will be pleased..

**Comment by azadvanshika** [comment]
hey, i m having problems with play protect and i have installed the secure version but still it says this device is not play protected

**Comment by afro972** [comment]
Hi
can i install it on my samsung S21 ULTRA SM-G9980 ? (snapdragon version for Hong Kong, Taiwan and China) 

thanks

**Reply by phhusson** [reply]
yes

**Reply by afro972** [reply]
ok thank you very much

**Comment by alexandreapis** [comment]
Some apps that plays videos starts to lag till crash, like twitter, tiktok... cant play more than 2 videos... but not happens on instagram or youtube, btw... Using on Xiaomi 11T Pro. Already tried disabling sound/audio/volume effects, enabled alternate audio policy, etc... Someone else experiencing it?

**Comment by wilsenbaltazar** [comment]
Can i install this AOSP on Galaxy A01? If yes, how can i do it?

**Reply by phhusson** [reply]
Yes you can, it's a tiny bit complicated to do, I can't explain it here. Best is you come to the telegram support group

**Reply by wilsenbaltazar** [reply]
Can you send me the telegram group link?

**Reply by pixapo** [reply]
I can't wake the device pushing the pwr button, the screen stay black. Also, the offline charging is not working, it keeps showing the battery icon while plugged in. Any solution?

**Comment by ilfrick** [comment]
Hello. I am on vndklite secure build, Xiaomi MI a2 lite. No camera at all, any way to fix this?
Thanks

**Reply by Aj-style** [reply]
hey didi you found any workaround for camera ?
